Can't change plan type for zone already active on Cloudflare

Hi there,

### Terraform Version
Terraform v0.12.23
+ provider.cloudflare v2.6.0

### Affected Resource(s)
cloudflare_zone

### Terraform Configuration Files
```
ariable "CLOUDFLARE_EMAIL" {}
variable "CLOUDFLARE_API_KEY" {}
variable "CLOUDFLARE_ACCOUNT_ID" {}
variable "CLOUDFLARE_ZONE_ID" {}
variable "CLOUDFLARE_ZONE" {}

provider "cloudflare" {
 email = var.CLOUDFLARE_EMAIL
 api_key = var.CLOUDFLARE_API_KEY
 account_id = var.CLOUDFLARE_ACCOUNT_ID
}

resource "cloudflare_zone" "terraform" {
  zone = var.CLOUDFLARE_ZONE
  plan = "enterprise"
}
```

### Debug Output
https://gist.github.com/jimbobjim123/8d0aae4352382d7b14bd03ede6e49676

### Expected Behavior
```
------------------------------------------------------------------------

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  # cloudflare_zone.terraform will be updated in-place
  ~ resource "cloudflare_zone" "terraform" {
        id                  = "69983a4885449aa792c10a62e6477f98"
        meta                = {
            "phishing_detected"  = "false"
            "wildcard_proxiable" = "false"
        }
        name_servers        = [
            "doug.ns.cloudflare.com",
            "jocelyn.ns.cloudflare.com",
        ]
        paused              = false
      ~ plan                = "free" -> "enterprise"
        status              = "active"
        type                = "full"
        vanity_name_servers = []
        zone                = "terraform.jamesaskham.us"
    }

Plan: 0 to add, 1 to change, 0 to destroy.

------------------------------------------------------------------------
```

Per the `terraform plan` the zone should be upgraded from Free to Enterprise plan

### Actual Behavior
```
Error: Error updating plan  for zone "69983a4885449aa792c10a62e6477f98": error from makeRequest: HTTP status 500: content "{\"success\":false,\"errors\":[{\"code\":1000,\"message\":\"Internal Server Error\"}]}\n"

  on cloudflare.tf line 13, in resource "cloudflare_zone" "terraform":
  13: resource "cloudflare_zone" "terraform" {
```

### Steps to Reproduce
Have an Active zone on Free plan imported into tfstate and then try to upgrade the zone. 

### Important Factoids
This works just fine when creating a new zone on the enterprise plan. We correctly send `CF_ENT` for the plan to the subscriptions API.

Didn't work, despite the API call now sending CF_ENT.
```
2020-04-27T14:09:57.043+0100 [DEBUG] plugin.terraform-provider-cloudflare: 2020/04/27 14:09:57 [DEBUG] Cloudflare API Request Details:
2020-04-27T14:09:57.043+0100 [DEBUG] plugin.terraform-provider-cloudflare: ---[ REQUEST ]---------------------------------------
2020-04-27T14:09:57.043+0100 [DEBUG] plugin.terraform-provider-cloudflare: PUT /client/v4/zones/fdd688f1f635dca79a46d234cad88985/subscription HTTP/1.1
2020-04-27T14:09:57.043+0100 [DEBUG] plugin.terraform-provider-cloudflare: Host: api.cloudflare.com
2020-04-27T14:09:57.043+0100 [DEBUG] plugin.terraform-provider-cloudflare: User-Agent: HashiCorp Terraform/0.12.23 (+https://www.terraform.io) Terraform Plugin SDK/1.10.0 terraform-provider-cloudflare/dev
2020-04-27T14:09:57.043+0100 [DEBUG] plugin.terraform-provider-cloudflare: Content-Length: 29
2020-04-27T14:09:57.043+0100 [DEBUG] plugin.terraform-provider-cloudflare: Content-Type: application/json
2020-04-27T14:09:57.043+0100 [DEBUG] plugin.terraform-provider-cloudflare: X-Auth-Email: jaskham@cloudflare.com
2020-04-27T14:09:57.043+0100 [DEBUG] plugin.terraform-provider-cloudflare: X-Auth-Key: 
2020-04-27T14:09:57.043+0100 [DEBUG] plugin.terraform-provider-cloudflare: Accept-Encoding: gzip
2020-04-27T14:09:57.043+0100 [DEBUG] plugin.terraform-provider-cloudflare:
2020-04-27T14:09:57.043+0100 [DEBUG] plugin.terraform-provider-cloudflare: {
2020-04-27T14:09:57.043+0100 [DEBUG] plugin.terraform-provider-cloudflare:  "rate_plan": {
2020-04-27T14:09:57.043+0100 [DEBUG] plugin.terraform-provider-cloudflare:   "id": "CF_ENT"
dflare:  }
2020-04-27T14:09:57.043+0100 [DEBUG] plugin.terraform-provider-cloudflare: }
2020-04-27T14:09:57.043+0100 [DEBUG] plugin.terraform-provider-cloudflare: -----------------------------------------------------
2020-04-27T14:09:57.688+0100 [DEBUG] plugin.terraform-provider-cloudflare: 2020/04/27 14:09:57 [DEBUG] Cloudflare API Response Details:
2020-04-27T14:09:57.688+0100 [DEBUG] plugin.terraform-provider-cloudflare: ---[ RESPONSE ]--------------------------------------
2020-04-27T14:09:57.688+0100 [DEBUG] plugin.terraform-provider-cloudflare: HTTP/1.1 400 Bad Request
2020-04-27T14:09:57.688+0100 [DEBUG] plugin.terraform-provider-cloudflare: Connection: close
2020-04-27T14:09:57.688+0100 [DEBUG] plugin.terraform-provider-cloudflare: Transfer-Encoding: chunked
2020-04-27T14:09:57.688+0100 [DEBUG] plugin.terraform-provider-cloudflare: Cf-Cache-Status: DYNAMIC
2020-04-27T14:09:57.688+0100 [DEBUG] plugin.terraform-provider-cloudflare: Cf-Ray: 58a8c7ea0806e60c-LHR
2020-04-27T14:09:57.688+0100 [DEBUG] plugin.terraform-provider-cloudflare: Cf-Request-Id: 025d5b46440000e60c390bb200000001
2020-04-27T14:09:57.688+0100 [DEBUG] plugin.terraform-provider-cloudflare: Content-Type: application/json; charset=utf-8
2020-04-27T14:09:57.688+0100 [DEBUG] plugin.terraform-provider-cloudflare: Date: Mon, 27 Apr 2020 13:09:57 GMT
2020-04-27T14:09:57.688+0100 [DEBUG] plugin.terraform-provider-cloudflare: Expect-Ct: max-age=604800, report-uri="https://report-uri.cloudflare.com/cdn-cgi/beacon/expect-ct"
2020-04-27T14:09:57.688+0100 [DEBUG] plugin.terraform-provider-cloudflare: Server: cloudflare
2020-04-27T14:09:57.688+0100 [DEBUG] plugin.terraform-provider-cloudflare: Set-Cookie: __cfduid=d335399e8fc3c288843eb7a6f6c3e27ce1587992997; expires=Wed, 27-May-20 13:09:57 GMT; path=/; domain=.api.cloudflare.com; HttpOnly; SameSite=Lax
2020-04-27T14:09:57.688+0100 [DEBUG] plugin.terraform-provider-cloudflare: Vary: Accept-Encoding
2020-04-27T14:09:57.688+0100 [DEBUG] plugin.terraform-provider-cloudflare: X-Envoy-Upstream-Service-Time: 91
2020-04-27T14:09:57.688+0100 [DEBUG] plugin.terraform-provider-cloudflare:
2020-04-27T14:09:57.688+0100 [DEBUG] plugin.terraform-provider-cloudflare: 84
2020-04-27T14:09:57.688+0100 [DEBUG] plugin.terraform-provider-cloudflare: {
2020-04-27T14:09:57.688+0100 [DEBUG] plugin.terraform-provider-cloudflare:  "success": false,
2020-04-27T14:09:57.688+0100 [DEBUG] plugin.terraform-provider-cloudflare:  "errors": [
2020-04-27T14:09:57.688+0100 [DEBUG] plugin.terraform-provider-cloudflare:   {
2020-04-27T14:09:57.688+0100 [DEBUG] plugin.terraform-provider-cloudflare:    "code": 1000,
2020-04-27T14:09:57.688+0100 [DEBUG] plugin.terraform-provider-cloudflare:    "message": "status: 400, message: subscription of that type does not exist for this zone"
2020-04-27T14:09:57.688+0100 [DEBUG] plugin.terraform-provider-cloudflare:   }
2020-04-27T14:09:57.688+0100 [DEBUG] plugin.terraform-provider-cloudflare:  ]
2020-04-27T14:09:57.689+0100 [DEBUG] plugin.terraform-provider-cloudflare: }
2020-04-27T14:09:57.689+0100 [DEBUG] plugin.terraform-provider-cloudflare:
2020-04-27T14:09:57.689+0100 [DEBUG] plugin.terraform-provider-cloudflare: 0
2020-04-27T14:09:57.689+0100 [DEBUG] plugin.terraform-provider-cloudflare:
2020-04-27T14:09:57.689+0100 [DEBUG] plugin.terraform-provider-cloudflare:
2020-04-27T14:09:57.689+0100 [DEBUG] plugin.terraform-provider-cloudflare: -----------------------------------------------------
2020/04/27 14:09:57 [DEBUG] cloudflare_zone.terraform: apply errored, but we're indicating that via the Error pointer rather than returning it: Error updating plan enterprise for zone "fdd688f1f635dca79a46d234cad88985": error from makeRequest: HTTP status 400: content "{\"success\":false,\"errors\":[{\"code\":1000,\"message\":\"status: 400, message: subscription of that type does not exist for this zone\"}]}\n"
2020/04/27 14:09:57 [ERROR] <root>: eval: *terraform.EvalApplyPost, err: Error updating plan enterprise for zone "fdd688f1f635dca79a46d234cad88985": error from makeRequest: HTTP status 400: content "{\"success\":false,\"errors\":[{\"code\":1000,\"message\":\"status: 400, message: subscription of that type does not exist for this zone\"}]}\n"
2020/04/27 14:09:57 [ERROR] <root>: eval: *terraform.EvalSequence, err: Error updating plan enterprise for zone "fdd688f1f635dca79a46d234cad88985": error from makeRequest: HTTP status 400: content "{\"success\":false,\"errors\":[{\"code\":1000,\"message\":\"status: 400, message: subscription of that type does not exist for this zone\"}]}\n"

Error: Error updating plan enterprise for zone "fdd688f1f635dca79a46d234cad88985": error from makeRequest: HTTP status 400: content "{\"success\":false,\"errors\":[{\"code\":1000,\"message\":\"status: 400, message: subscription of that type does not exist for this zone\"}]}\n"
```
Inspecting this in the dashboard when I upgrade a zone, it actually looks like this call works: 

```
curl --location --request PATCH 'https://api.cloudflare.com/client/v4/zones/fdd688f1f635dca79a46d234cad88985/' \
--header 'X-Auth-Email: jaskham@cloudflare.com' \
--header 'X-Auth-Key: ' \
--header 'Content-Type: application/json' \
--header 'Content-Type: text/plain' \
--header 'Cookie: __cfduid=dccc50936f2f9e315fa2cdb9d91c664791563808220' \
--data-raw '{  
    "plan":{  
        "id":"94f3b7b768b0458b56d2cac4fe5ec0f9"
    }
}'
```
Unsure if this is an API that is supported for use in this way or if the available plan id's are unique per account?

--- a/cloudflare/resource_cloudflare_zone.go
+++ b/cloudflare/resource_cloudflare_zone.go
@@ -217,15 +217,15 @@ func resourceCloudflareZoneUpdate(d *schema.ResourceData, meta interface{}) erro
 	}
 
 	// In the cases where the zone isn't completely setup yet, we need to
-	// check the `status` field and should it be pending, use the `Name`
+	// check the `status` field and should it be pending, use the `LegacyID`
 	// from `zone.PlanPending` instead to account for paid plans.
 	if zone.Status == "pending" && zone.PlanPending.Name != "" {
-		d.Set("plan", zone.PlanPending.Name)
+		d.Set("plan", zone.PlanPending.LegacyID)
 	}
 
 	if plan, ok := d.GetOk("plan"); ok {
-		planName := planIDForName(plan.(string))
-		if err := setRatePlan(client, zoneID, planName, false); err != nil {
+		planID := plan.(string)
+		if err := setRatePlan(client, zoneID, planID, false); err != nil {
 			return err
 		}
 	}

--- a/cloudflare/resource_cloudflare_zone.go
+++ b/cloudflare/resource_cloudflare_zone.go
@@ -223,13 +223,13 @@ func resourceCloudflareZoneUpdate(d *schema.ResourceData, meta interface{}) erro
 		d.Set("plan", zone.PlanPending.LegacyID)
 	}
 
-	if plan, ok := d.GetOk("plan"); ok {
+	if change := d.HasChange("plan"); change {
 		// If we're upgrading from a free plan, we need to use POST (not PUT) as the
 		// the subscription needs to be created, not modified despite the resource
 		// already existing.
-		existingPlan, _ := d.GetChange("plan")
+		existingPlan, newPlan := d.GetChange("plan")
 		wasFreePlan := existingPlan.(string) == "free"
-		planID := plan.(string)
+		planID := newPlan.(string)
 
 		if err := setRatePlan(client, zoneID, planID, wasFreePlan); err != nil {
 			return err

--- a/cloudflare/resource_cloudflare_zone.go
+++ b/cloudflare/resource_cloudflare_zone.go
@@ -224,8 +224,14 @@ func resourceCloudflareZoneUpdate(d *schema.ResourceData, meta interface{}) erro
 	}
 
 	if plan, ok := d.GetOk("plan"); ok {
+		// If we're upgrading from a free plan, we need to use POST (not PUT) as the
+		// the subscription needs to be created, not modified despite the resource
+		// already existing.
+		existingPlan, _ := d.GetChange("plan")
+		wasFreePlan := existingPlan.(string) == "free"
 		planID := plan.(string)
-		if err := setRatePlan(client, zoneID, planID, false); err != nil {
+
+		if err := setRatePlan(client, zoneID, planID, wasFreePlan); err != nil {
 			return err
 		}
 	}

